react/src/components/atoms/forms/InputCheckBox/index.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Input from '../Input';
+import Error from '../Input/error';
+import Icon from '../../icons/Icon';
+import { InputContext } from '../Input/context';
+import './style.css';
+
+const CheckBox = (props) => (
+  <React.Fragment>
+    <InputContext.Consumer>
+      {
+        (context) => {
+          const { id, value } = context;
+          const {
+            icon, label, disabled, required
+          } = props;
+          const handleClick = () => {
+            context.updateState({ value: !value }, () => {
+              if (typeof props.onChange === 'function') {
+                props.onChange(context.getValue(), id);
+              }
+            });
+            if (!!value && required) {
```

Comment:
I am no seeing required do anything here....

Reply:
if you check required and not check the input checkbox:

![screen shot 2019-02-13 at 2 45 55 pm](https://user-images.githubusercontent.com/5789411/52739066-1975ce80-2f9e-11e9-8586-63edc22f9f8e.png)